lib/tasks/after_tasks.rake

```diff
@@ -740,6 +740,34 @@ namespace :After do
     WorkIndexer.create_mapping
   end
 
+  desc "Fix tags with extra spaces"
+  task(fix_tags_with_extra_spaces: :environment) do
+    total_tags = Tag.count
+    total_batches = (total_tags + 999) / 1000
+    puts "Inspecting #{total_tags} tags in #{total_batches} batches"
+
+    report_string = "Tag ID,Old tag name,New tag name\n"
+    Tag.find_in_batches.with_index do |batch, index|
+      batch_number = index + 1
+      progress_msg = "Batch #{batch_number} of #{total_batches} complete"
+
+      batch.each do |tag|
+        next unless tag.name != tag.name.squish
+
+        old_tag_name = tag.name
+        new_tag_name = old_tag_name.gsub(/[[:space:]]/, "_")
+
+        tag.update_column(:name, new_tag_name)
```

Comment:
What happens if there's already a version of the tag with the underscores in it?

Also, `update_column` skips callbacks like reindexing and cache expiration. Maybe it'd be good to use `update_attribute` instead? I think that skips validations while still performing callbacks.

Reply:
The task will fail if there is an existing tag already with the underscores in it the name clashes.  Rails is skipping validations but there is still a unique constraint in the DB table and it will fail to save the tag.

Not sure what's the desired behaviour in this case, abort?  Skip and log?  Keep adding underscores (or some other prefix/suffix) to make it unique?

Reply:
Let's keep adding underscores, yeah.